File: C++/0014-Longest-Common-Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string longestPrefix = "";
        string longestPrefixTemp = "";
        if (strs.size() == 1) {
            longestPrefix = strs[0];
        }
        else {
            string temp1 = strs[0];
            string temp2 = strs[1];
            
            for (int i = 0; i < temp1.length() && i < temp2.length(); i++) {
                if (temp1[i] == temp2[i]) {
                    longestPrefixTemp += temp1[i];
                }
                else {break;}
            }
            longestPrefix = longestPrefixTemp;
            longestPrefixTemp = "";
            
            for (int i = 1; i < strs.size(); i++) {
                temp2 = strs[i];
                for (int j = 0; j < longestPrefix.length() && j < temp2.length(); j++) {
                    if (longestPrefix[j] == temp2[j]) {
                        longestPrefixTemp += longestPrefix[j];
                    }
                    else {break;}
                }
                longestPrefix = longestPrefixTemp;
                longestPrefixTemp = "";
            }
        }
        return longestPrefix;
    }
};

File: Java/0014-Longest-Common-Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix = "";
        String check = "";
        for (int i = 0; i < strs[0].length(); i++) {
            check = prefix + strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].indexOf(check) == -1 || strs[j].indexOf(check) != 0) {
                    return prefix;
                }
            }
            prefix = check;
        }

        return prefix;
    }
}